include/irr/asset/IImageLoader.h

```diff
@@ -33,29 +33,26 @@ namespace irr
 
 				uint8_t* convertR8SRGBdataIntoRGB8SRGBAAndGetIt(const void* redChannelDataLayer, const core::smart_refctd_ptr<ICPUImage>& image, const irr::asset::IImage::SBufferCopy& region)
 				{
-					constexpr auto channelAmount = 3;
-					const void* planarData[] = { redChannelDataLayer, nullptr, nullptr, nullptr };
-					uint8_t* out = nullptr;
-					const size_t texelAmount = getFormatChannelCount(image->getCreationParameters().format);
-
 					const auto& imageSize = image->getCreationParameters().extent;
 					irr::core::vector3d<uint32_t> imageSizeWithPitch;
 					imageSizeWithPitch.X = region.bufferRowLength > 0 ? region.bufferRowLength : region.imageExtent.width;
 					imageSizeWithPitch.Y = region.bufferImageHeight > 0 ? region.bufferImageHeight : region.imageExtent.height;
 					imageSizeWithPitch.Z = region.imageExtent.depth;
-
-					video::convertColor<EF_R8_SRGB, EF_R8G8B8_SRGB>(planarData, out, texelAmount, imageSizeWithPitch);
+					
+					constexpr auto pixelByteSize = 3;
+					const void* planarData[] = { redChannelDataLayer, nullptr, nullptr, nullptr };
+					uint8_t* out = _IRR_NEW_ARRAY(uint8_t, imageSizeWithPitch.X * imageSizeWithPitch.Y * imageSizeWithPitch.Z);
```

Comment:
don't allocate memory in the function! just take an `ICPUBuffer` to go with the `region` and use that.